server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1340,8 +1348,8 @@ private void doGetRepositoryData(ActionListener<RepositoryData> listener) {
                     // We can cache serialized in the most recent version here without regard to the actual repository metadata version
                     // since we're only caching the information that we just wrote and thus won't accidentally cache any information that
                     // isn't safe
-                    cacheRepositoryData(
-                            BytesReference.bytes(loaded.snapshotsToXContent(XContentFactory.jsonBuilder(), Version.CURRENT)), genToLoad);
+                    cacheRepositoryData(compressRepoDataForCache(BytesReference.bytes(
```

Comment:
It's not exclusively laziness that I didn't make this path use pooled bytes now as well. Caching here turned out to be somewhat bogus in practice and needs to be changed (will do over the next few days when I find some time). I just went with the simplest possible change here for now as it doesn't really matter.